### PR TITLE
Reference redirectors and jenkins.io as much as possible

### PR DESCRIPTION
**Pull request by daniel-beck**
In the process, I made the redirector URLs more specific to more easily adapt to documentation format changes (e.g. splitting up documentation into multiple pages).

Example: http://wiki.jenkins-ci.org/display/JENKINS/Administering+Jenkins → `/redirect/migrate-jenkins-home`

* Plugin wiki pages were replaced by plugins.jenkins.io which at least as a redirector service can run indefinitely.
* References to JENKINS issues in Javadoc were replaced by the jenkins-ci.org/issue/1234 redirector
* References to other wiki pages were replaced by general `/redirect` (https://github.com/jenkins-infra/jenkins.io/tree/master/content/redirect) URLs.

As a side effect of the use of redirect URLs, references to the Japanese wiki will in the future (until Accept-Language support on the site is implemented in some way) point to the English wiki/documentation.

Minor adaptations besides pure wiki/issue tracker URL rewrites:

* I replaced some occurrences of "wiki" or "wiki page" with "documentation". There's tons left though, and I didn't even touch languages I don't speak.
* I replaced an occurrence of "cloudbees.com" in user help with "jenkins.io".
* I removed the `ciManagement` entry from `pom.xml`. It was seriously outdated anyway.
* I changed the changelog URLs shown when a new version is available and defined in pom.xml with the jenkins.io variants.

### CODE REVIEW

**Comment by daniel-beck** [comment]
@olivergondza @jglick Any objections on the basis of backporting/merging?

File: cli/src/main/resources/hudson/cli/client/Messages.properties [hunk 1]
```diff
@@ -3,7 +3,7 @@ CLI.Usage=Jenkins CLI\n\
   Options:\n\
    -s URL       : the server URL (defaults to the JENKINS_URL env var)\n\
    -i KEY       : SSH private key file used for authentication\n\
-   -p HOST:PORT : HTTP proxy host and port for HTTPS proxy tunneling. See http://jenkins-ci.org/https-proxy-tunnel\n\
```
Comment (daniel-beck) [inline cli/src/main/resources/hudson/cli/client/Messages.properties]: @kohsuke Apparently ([WEBSITE-319](https://issues.jenkins-ci.org/browse/WEBSITE-319)) this URL never existed. Any idea where this was supposed to go? Or did this documentation never exist?

**Comment by daniel-beck** [comment]
@olivergondza @jglick PTAL.

**Comment by olivergondza** [comment]
This looks good to me after a brief scan.

**Comment by daniel-beck** [comment]
Changed a few URLs around that I added in https://github.com/jenkins-infra/jenkins.io/pull/695

File: README.md [hunk 2]
```diff
@@ -59,6 +59,4 @@ Jenkins is **licensed** under the **[MIT License]**. The terms of the license ar
 [GitHub]: https://github.com/jenkinsci/jenkins
 [website]: https://jenkins.io/
 [@jenkinsci]: https://twitter.com/jenkinsci
-[Contributing]: https://wiki.jenkins-ci.org/display/JENKINS/contributing
```
Comment (oleg-nenashev) [inline README.md]: Maybe a link to the file within repo>

File: core/src/main/java/hudson/Proc.java [hunk 3]
```diff
@@ -503,7 +503,7 @@ public OutputStream getStdin() {
     /**
     * An instance of {@link Proc}, which has an internal workaround for JENKINS-23271.
     * It presumes that the instance of the object is guaranteed to be used after the {@link Proc#join()} call.
-    * See <a href="https://issues.jenkins-ci.org/browse/JENKINS-23271">JENKINS-23271></a>
+    * See <a href="https://jenkins-ci.org/issue/23271">JENKINS-23271></a>
```
Comment (oleg-nenashev) [inline core/src/main/java/hudson/Proc.java]: I am fine with that, but maybe makes no sense without jenkins.io

File: core/src/main/resources/hudson/ProxyConfiguration/help-noProxyHost.html [hunk 4]
```diff
@@ -1,4 +1,4 @@
 <div>
     Specify host name patterns that shouldn't go through the proxy, one host per line.
-    "*" is the wild card host name (such as "*.cloudbees.com" or "www*.jenkins-ci.org")
+    "*" is the wild card host name (such as "*.jenkins.io" or "www*.jenkins-ci.org")
```
Comment (oleg-nenashev) [inline core/src/main/resources/hudson/ProxyConfiguration/help-noProxyHost.html]: Uh oh, thanks for cleaning it up.
Maybe the second example with `www*.jenkins-ci.org` should be somehow replaced as well

**Review by oleg-nenashev — APPROVED** [review]
I am not exactly sure about usefulness of JIRA redirector since it uses `jenkins-ci.org`, but it could be fixed during the next iteration. 👍 for the merge

File: README.md (hunk 2 above)
Comment (daniel-beck) [inline README.md]: @oleg-nenashev https://github.com/jenkinsci/jenkins/pull/2756/files#diff-04c6e90faac2675aa89e2176d2eec7d8L26

**Comment by daniel-beck** [comment]
@oleg-nenashev 

> JIRA redirector

Yeah, both options we have aren't great right now. FWIW I asked @rtyler which he preferred, and he said in the case of Javadoc JIRA issue references, he doesn't care. User-visible references should be removed though.

**Comment by daniel-beck** [comment]
https://github.com/jenkins-infra/jenkins.io/pull/695 has been merged, so this can now be merged as well.

File: README.md (hunk 2 above)
Comment (oleg-nenashev) [inline README.md]: Agreed

**Pull request merged**